test/parallel_api/numeric/numeric.ops/inclusive_scan_by_segment_zip.pass.cpp

```diff
@@ -0,0 +1,112 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "support/test_config.h"
+
+#include _PSTL_TEST_HEADER(execution)
+
+#include "support/utils.h"
+
+#include _PSTL_TEST_HEADER(algorithm)
+#include _PSTL_TEST_HEADER(iterator)
+#include _PSTL_TEST_HEADER(functional)
+
+#include <functional>
+#include <iostream>
+
+#if TEST_DPCPP_BACKEND_PRESENT
+#include "support/sycl_alloc_utils.h"
+
+template <sycl::usm::alloc alloc_type, typename KernelName>
+void
+test_with_usm()
+{
+    sycl::queue q = TestUtils::get_test_queue();
+
+    constexpr int n = 9;
+
+    //data initialization
+    int keys1  [n] = { 11, 11, 21, 20, 21, 21, 21, 37, 37 };
+    int keys2  [n] = { 11, 11, 20, 20, 20, 21, 21, 37, 37 };
+    int values1[n] = {  0,  1,  2,  3,  4,  5,  6,  7,  8 };
+    int values2[n] = {  0,  1,  2,  3,  4,  5,  6,  7,  8 };
+    int output_values1[n] = { };
+    int output_values2[n] = { };
+
+    // allocate USM memory and copying data to USM shared/device memory
+    TestUtils::usm_data_transfer<alloc_type, int> dt_helper1(q, std::begin(keys1),          std::end(keys1));
```

Comment:
May be better to use something like ```TestUtils::usm_data_transfer<alloc_type, int> dt_helper1(q, keys1, n);``` ?

Reply:
Yes, this syntax is better. I have made this change.